tests/FsCheck.Test/Gen.fs

```diff
@@ -181,6 +182,29 @@ module Gen =
             && (Array2D.length2 arr <= cols) 
             && (seq { for elem in arr do yield elem :?> int} |> Seq.forall ((=) v))
 
+
+    type MaybeNull =
+        {
+            SomeField : int
+        }
+
+    [<CLIMutable>]
+    type MaybeHaveNull =
+        {
+             MightBeNull : MaybeNull
+        }
+
+    [<Fact>]
+    let ``should support null as a shrunken value even without AllowNullLiteral (for C# compatibility)``() =
+        let der = Arb.Default.Derive()
+        let mhn = { MightBeNull = { SomeField = 5 } }
+        let prop = mhn.GetType().GetProperty("MightBeNull", BindingFlags.Public ||| BindingFlags.Instance)
+        prop.SetValue(mhn, null)
+        // ensure that no exception occurs
+        der.Shrinker mhn
+          |> Seq.length
+          |> ignore
```

Comment:
I may be missing something, but this seems like a very roundabout way of testing the change. How about something like:

     let shrinker = Arb.Default.Derive().Shrinker
     shrinker null |> Seq.length != 0